### PR TITLE
Catch most SPICE errors

### PR DESCRIPTION
**Pull request by Schneegans**
This fixes several SPICE-related crashes.

This fixes issue #102

### CODE REVIEW

File: plugins/csp-anchor-labels/src/AnchorLabel.cpp [hunk 1]
```diff
@@ -104,34 +106,40 @@ void AnchorLabel::update() {
...
 
-    auto rot = glm::toQuat(glm::dmat3(x, y, z));
-    mAnchor->setAnchorRotation(rot);
+      mAnchor->update(simulationTime, mSolarSystem->getObserver());
 
-    mAnchor->update(simulationTime, mSolarSystem->getObserver());
+    } catch (std::exception const& e) {
+      // Getting the relative transformation may fail due to insufficient SPICE data.
+      logger().warn("AnchorLabel::update failed for '{}': {}", mBody->getCenterName(), e.what());
+    }
```
Comment (JonasGilg) [inline plugins/csp-anchor-labels/src/AnchorLabel.cpp:142]: While this prevents the application from a crash, it leaves it in an almost unusable state. When you follow MRO until its frame expires no more movement is possible. The only recovery possible is to revert time to a point where MRO existed and then move out of frame.

A better solution would be to change the center and frame once this happens.

**Review by JonasGilg — CHANGES_REQUESTED** [review]

**Comment by Schneegans** [comment]
Soo... can you test this once more? Simply changing the SPICE frame is obviously impossible as we do not have enough data to compute the position / orientation of it in the first place. I have now implmented some kind of workaround which keeps track of the last known working simulation time and uses this in case of an error to reset the observer frame. This seems to work but feels kind of fragile. At least, the application is still usable if the current SPICE frame fails to update....

File: src/cs-core/SolarSystem.cpp [hunk 2]
```diff
@@ -89,6 +89,25 @@ scene::CelestialObserver const& SolarSystem::getObserver() const {
...
+  try {
+    mObserver.getRelativePosition(mTimeControl->pSimulationTime.get(),
+        scene::CelestialAnchor("Solar System Barycenter", "J2000"));
+  } catch (...) {
+    // In case of an error, we reset the observer. This can throw an error itself, but we cannot do
+    // anything if that happens.
+    try {
+      mObserver.changeOrigin("Solar System Barycenter", "J2000", lastWorkingSimulationTime);
+    } catch (std::exception const& e) {
+      logger().warn("Failed to reset the Observer SPICE frame: {}", e.what());
```
Comment (JonasGilg) [inline src/cs-core/SolarSystem.cpp:104]: If this happens, shouldn't that be an error or critical error, since the application state is not recoverable?

**Review by JonasGilg — CHANGES_REQUESTED** [review]
While it might be hacky, it is much better than having a broken app state. From a user perspective it actually feels like intended behaviour.

File: src/cs-core/SolarSystem.cpp (hunk 2 above)
Comment (Schneegans) [reply src/cs-core/SolarSystem.cpp]: Good point!

**Review by JonasGilg — APPROVED** [review]

**Pull request merged**